Update project.service.ts

AkRao2705 opened the pull request:
Updated project.service.ts to allow the uploaded document to be deleted from the database when the delete option is selected.

AkRao2705 committed:
Update project.service.ts

Updated project.service.ts to allow the uploaded document to be deleted from the database when the delete option is selected.

Keith3895 commented on src/services/project.service.ts:
```diff
@@ -1,18 +1,29 @@
+
+  
+  constructor() {
+    if (!this.projectRepository) {
```
why do we need this block?

Keith3895 commented on src/services/project.service.ts:
```diff
@@ -33,8 +44,61 @@ export class ProjectService {
+                    logger.error('Error deleting file:', err);
+                  } else {
+                    logger.info('File deleted successfully:', filePath);
+                    console.log(`File ${filePath} deleted successfully from local storage.`);
```
you can remove the console.log

Keith3895 commented on src/services/project.service.ts:
```diff
@@ -33,8 +44,61 @@ export class ProjectService {
+
+      // Delete the documents from the database
+      await Promise.all(
+        project.documents.map(async (document) => {
```
instead of looping over an array you can just delete where document in the list. check if its possible.

Keith3895 reviewed (changes requested):
add ed comments.

AkRao2705 committed:
Update project.service.ts

Removed constructor block which initialized project repository (NOT NEEDED). 
Instead of deleting each document one by one, we now delete all documents at once using their IDs in a single query.

Keith3895 reviewed (changes requested):
update to use cloud storage logic.

AkRao2705 committed:
Create storage.service.ts

Adds a StorageService to handle Google Cloud Storage file operations (deleting files).

Modifies ProjectService to use the StorageService when deleting associated files along with the project.

AkRao2705 committed:
Update project.service.ts

Updated project.service.ts by including cloud logic.

AkRao2705 committed:
Merge branch 'main' into delete-project-feature
